source/common/upstream/cluster_manager_impl.cc

```diff
@@ -1474,6 +1452,38 @@ ClusterManagerImpl::ThreadLocalClusterManagerImpl::ClusterEntry::connPool(
   }
 }
 
+void ClusterManagerImpl::ThreadLocalClusterManagerImpl::ClusterEntry::httpConnPoolIsIdle(
+    HostConstSharedPtr host, ResourcePriority priority, const std::vector<uint8_t>& hash_key) {
+  if (parent_.destroying_) {
```

Comment:
optional, think it's worth a templated helper here?

Reply:
I think the two methods are different enough that it will make it harder to read/understand. There's a bunch of differences between how we manage http vs tcp pools, that probably shouldn't be different, but they are.